packages/picasso.js/src/web/text-manipulation/text-metrics.js

```diff
@@ -17,21 +17,31 @@ function setContext() {
   context = context || document.createElement('canvas').getContext('2d');
 }
 
-function setFont(fontSize, fontFamily) {
-  if (contextCache.fontSize === fontSize && contextCache.fontFamily === fontFamily) {
+function isEqual(s1, s2) {
+  return (!s1 && !s2) || s1 === s2;
+}
+
+function setFont({ fontStyle, fontWeight, fontSize, fontFamily }) {
+  if (
+    isEqual(contextCache.fontStyle, fontStyle) &&
+    isEqual(contextCache.fontWeight, fontWeight) &&
+    isEqual(contextCache.fontSize, fontSize) &&
+    isEqual(contextCache.fontFamily, fontFamily)
+  ) {
     return;
   }
-
-  context.font = fontSize + ' ' + fontFamily; // eslint-disable-line
+  context.font = [fontStyle, fontWeight, fontSize, fontFamily].filter((value) => !!value).join(' '); // eslint-disable-line
+  contextCache.fontStyle = fontStyle;
+  contextCache.fontWeight = fontWeight;
   contextCache.fontSize = fontSize;
   contextCache.fontFamily = fontFamily;
 }
 
-function measureTextWidth(text, fontSize, fontFamily) {
-  const key = text + fontSize + fontFamily;
+function measureTextWidth({ text, fontStyle, fontWeight, fontSize, fontFamily }) {
+  const key = text + [fontStyle, fontWeight, fontSize, fontFamily].filter((value) => !!value).join('');
```

Comment:
Since `measureTextWidth` is expected to be called A LOT. If I recall correctly, given a text string `Some text`, `measureTextWidth` is called for `S`, `So`, `Som`, `Some` and so forth.

So we should probably avoid running a filter function here. Since it's also executed at line 33, although not as often.

Reply:
What is your solution then?

Reply:
Is this better 
`[fontStyle || '', fontWeight || '', fontSize || '', fontFamily || ''].join('');`

Reply:
A string literal? Assuming that is less costly.

```js
const key = `${text} ${fontStyle} ${fontWeight} ${fontSize} ${fontFamily}`
```

Reply:
Updated!